### PR TITLE
Add check destroy for aws ebs snapshot copy resource & Fix error handling

### PR DESCRIPTION
**Pull request by kterada0509**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" comments, they generate extra noise for pull request followers and do not help prioritize the request

References #8958 

Release note for [CHANGELOG](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md):

```release-note
BUG-FIX:
- resource/aws_ebs_snapshot_copy: fix error handling when resource is gone
```

Output from acceptance testing:

```
$ make testacc TESTARGS="-run=TestAccAWSEbsSnapshotCopy_"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./... -v -parallel 20 -run=TestAccAWSEbsSnapshotCopy_ -timeout 120m
?   	github.com/terraform-providers/terraform-provider-aws	[no test files]
=== RUN   TestAccAWSEbsSnapshotCopy_basic
=== PAUSE TestAccAWSEbsSnapshotCopy_basic
=== RUN   TestAccAWSEbsSnapshotCopy_withDescription
=== PAUSE TestAccAWSEbsSnapshotCopy_withDescription
=== RUN   TestAccAWSEbsSnapshotCopy_withRegions
=== PAUSE TestAccAWSEbsSnapshotCopy_withRegions
=== RUN   TestAccAWSEbsSnapshotCopy_withKms
=== PAUSE TestAccAWSEbsSnapshotCopy_withKms
=== RUN   TestAccAWSEbsSnapshotCopy_disappears
=== PAUSE TestAccAWSEbsSnapshotCopy_disappears
=== CONT  TestAccAWSEbsSnapshotCopy_basic
=== CONT  TestAccAWSEbsSnapshotCopy_disappears
=== CONT  TestAccAWSEbsSnapshotCopy_withKms
=== CONT  TestAccAWSEbsSnapshotCopy_withDescription
=== CONT  TestAccAWSEbsSnapshotCopy_withRegions
--- PASS: TestAccAWSEbsSnapshotCopy_disappears (73.88s)
--- PASS: TestAccAWSEbsSnapshotCopy_withDescription (74.98s)
--- PASS: TestAccAWSEbsSnapshotCopy_withKms (101.45s)
--- PASS: TestAccAWSEbsSnapshotCopy_basic (105.72s)
--- PASS: TestAccAWSEbsSnapshotCopy_withRegions (109.75s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	109.851s
```

### CODE REVIEW

File: aws/resource_aws_ebs_snapshot_copy.go [hunk 1]
```diff
@@ -116,7 +116,7 @@ func resourceAwsEbsSnapshotCopyRead(d *schema.ResourceData, meta interface{}) er
 		SnapshotIds: []*string{aws.String(d.Id())},
 	}
 	res, err := conn.DescribeSnapshots(req)
-	if ec2err, ok := err.(awserr.Error); ok && ec2err.Code() == "InvalidSnapshotID.NotFound" {
+	if ec2err, ok := err.(awserr.Error); ok && ec2err.Code() == "InvalidSnapshot.NotFound" {
```
Comment (bflad) [inline aws/resource_aws_ebs_snapshot_copy.go]: These can be simplified using `isAWSErr(err, CODE, MESSAGE)` in the future, e.g.

```suggestion
	if isAWSErr(err, "InvalidSnapshot.NotFound", "") {
```

File: aws/resource_aws_ebs_snapshot_copy.go [hunk 2]
```diff
@@ -153,10 +153,14 @@ func resourceAwsEbsSnapshotCopyDelete(d *schema.ResourceData, meta interface{})
 		}
 
 		ebsErr, ok := err.(awserr.Error)
-		if ebsErr.Code() == "SnapshotInUse" {
+		if ok && ebsErr.Code() == "SnapshotInUse" {
```
Comment (bflad) [inline aws/resource_aws_ebs_snapshot_copy.go]: Leaving as non-blocking feedback for next time.

```suggestion
		if isAWSErr(err, "SnapshotInUse", "") {
```

File: aws/resource_aws_ebs_snapshot_copy.go [hunk 3]
```diff
@@ -153,10 +153,14 @@ func resourceAwsEbsSnapshotCopyDelete(d *schema.ResourceData, meta interface{})
...
 			return resource.RetryableError(fmt.Errorf("EBS SnapshotInUse - trying again while it detaches"))
 		}
 
+		if ok && ebsErr.Code() == "InvalidSnapshot.NotFound" {
```
Comment (bflad) [inline aws/resource_aws_ebs_snapshot_copy.go]:
```suggestion
		if isAWSErr(err, "InvalidSnapshot.NotFound", "") {
```

File: aws/resource_aws_ebs_snapshot_copy_test.go [hunk 4]
```diff
@@ -93,6 +98,67 @@ func TestAccAWSEbsSnapshotCopy_withKms(t *testing.T) {
...
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "aws_ebs_snapshot_copy" {
+			continue
+		}
+
+		resp, err := conn.DescribeSnapshots(&ec2.DescribeSnapshotsInput{
+			SnapshotIds: []*string{aws.String(rs.Primary.ID)},
+		})
+
+		if ec2err, ok := err.(awserr.Error); ok && ec2err.Code() == "InvalidSnapshot.NotFound" {
```
Comment (bflad) [inline aws/resource_aws_ebs_snapshot_copy_test.go]:
```suggestion
		if isAWSErr(err, "InvalidSnapshot.NotFound", "") {
```

File: aws/resource_aws_ebs_snapshot_copy_test.go [hunk 5]
```diff
@@ -93,6 +98,67 @@ func TestAccAWSEbsSnapshotCopy_withKms(t *testing.T) {
...
+			return nil
```
Comment (bflad) [inline aws/resource_aws_ebs_snapshot_copy_test.go]: `CheckDestroy` should not use `return nil` in the `for` loop, to ensure all resources are properly checked. 👍 

```suggestion
			continue
```

**Comment by kterada0509** [comment]
Re-run acctest.

```
$ make testacc TESTARGS="-run=TestAccAWSEbsSnapshotCopy_"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./... -v -parallel 20 -run=TestAccAWSEbsSnapshotCopy_ -timeout 120m
?   	github.com/terraform-providers/terraform-provider-aws	[no test files]
=== RUN   TestAccAWSEbsSnapshotCopy_basic
=== PAUSE TestAccAWSEbsSnapshotCopy_basic
=== RUN   TestAccAWSEbsSnapshotCopy_withDescription
=== PAUSE TestAccAWSEbsSnapshotCopy_withDescription
=== RUN   TestAccAWSEbsSnapshotCopy_withRegions
=== PAUSE TestAccAWSEbsSnapshotCopy_withRegions
=== RUN   TestAccAWSEbsSnapshotCopy_withKms
=== PAUSE TestAccAWSEbsSnapshotCopy_withKms
=== RUN   TestAccAWSEbsSnapshotCopy_disappears
=== PAUSE TestAccAWSEbsSnapshotCopy_disappears
=== CONT  TestAccAWSEbsSnapshotCopy_basic
=== CONT  TestAccAWSEbsSnapshotCopy_withRegions
=== CONT  TestAccAWSEbsSnapshotCopy_withKms
=== CONT  TestAccAWSEbsSnapshotCopy_withDescription
=== CONT  TestAccAWSEbsSnapshotCopy_disappears
--- PASS: TestAccAWSEbsSnapshotCopy_withDescription (83.19s)
--- PASS: TestAccAWSEbsSnapshotCopy_basic (83.71s)
--- PASS: TestAccAWSEbsSnapshotCopy_disappears (98.24s)
--- PASS: TestAccAWSEbsSnapshotCopy_withRegions (101.98s)
--- PASS: TestAccAWSEbsSnapshotCopy_withKms (125.85s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	125.991s
```

File: aws/resource_aws_ebs_snapshot_copy_test.go (hunk 5 above)
Comment (kterada0509) [reply aws/resource_aws_ebs_snapshot_copy_test.go]: fixed.

File: aws/resource_aws_ebs_snapshot_copy_test.go (hunk 4 above)
Comment (kterada0509) [reply aws/resource_aws_ebs_snapshot_copy_test.go]: fixed.

File: aws/resource_aws_ebs_snapshot_copy.go (hunk 3 above)
Comment (kterada0509) [reply aws/resource_aws_ebs_snapshot_copy.go]: fixed.

File: aws/resource_aws_ebs_snapshot_copy.go (hunk 2 above)
Comment (kterada0509) [reply aws/resource_aws_ebs_snapshot_copy.go]: fixed.

File: aws/resource_aws_ebs_snapshot_copy.go (hunk 1 above)
Comment (kterada0509) [reply aws/resource_aws_ebs_snapshot_copy.go]: fixed.

**Comment by bflad** [comment]
This has been released in [version 2.17.0 of the Terraform AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#2170-june-28-2019). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.

For further feature requests or bug reports with this functionality, please create a [new GitHub issue](https://github.com/terraform-providers/terraform-provider-aws/issues/new/choose) following the template for triage. Thanks!

**Pull request merged**